pkg/service/export.go

```diff
@@ -0,0 +1,96 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	autil "github.com/submariner-io/admiral/pkg/util"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+func Export(restConfigProducer restconfig.Producer, serviceNamespace, svcName string) error {
+	status := cli.NewReporter()
+
+	clientConfig := restConfigProducer.ClientConfig()
+
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		return status.Error(err, "Error connecting to the target cluster")
+	}
+
+	clientProducer, err := client.NewProducerFromRestConfig(restConfig)
+	if err != nil {
+		return status.Error(err, "Error creating client producer")
+	}
+
+	restMapper, err := autil.BuildRestMapper(restConfig)
+	if err != nil {
+		return status.Error(err, "Error creating RestMapper for ServiceExport")
+	}
+
+	if serviceNamespace == "" {
+		if serviceNamespace, _, err = clientConfig.Namespace(); err != nil {
+			serviceNamespace = "default"
+		}
+	}
+
+	err = mcsv1a1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return status.Error(err, "Failed to add to scheme")
+	}
+
+	_, err = clientProducer.ForKubernetes().CoreV1().Services(serviceNamespace).Get(context.TODO(), svcName, metav1.GetOptions{})
+	if err != nil {
+		return status.Error(err, "Unable to find the Service %q in namespace %q", svcName, serviceNamespace)
+	}
+
+	mcsServiceExport := &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcName,
+			Namespace: serviceNamespace,
+		},
+	}
+
+	resourceServiceExport, gvr, err := autil.ToUnstructuredResource(mcsServiceExport, restMapper)
+	if err != nil {
+		return status.Error(err, "Failed to convert to Unstructured")
+	}
```

Comment:
We don't really need to dynamically find the GVR since we know that info directly. Hence we don't need to build a `RESTMapper` and thus we don't need a `rest.Config` which simplifies the code. 

```suggestion
        import ""github.com/submariner-io/admiral/pkg/resource"
        ...
	resourceServiceExport, err := resource.ToUnstructured(mcsServiceExport)
	if err != nil {
		return status.Error(err, "Failed to convert to Unstructured")
	}
	
	gvr := &schema.GroupVersionResource{
		Group:    mcsv1a1.GroupVersion.Group,
		Version:  mcsv1a1.GroupVersion.Version,
		Resource: "serviceexports",
	}
```